src/test/java/server/user/service/UserServiceTest.java

```diff
@@ -0,0 +1,189 @@
+import server.user.dto.request.UserPostRequest;
+import server.user.dto.request.UserPutRequest;
+import server.user.entity.Badge;
+import server.user.entity.User;
+import server.user.fixture.UserFixture;
+import server.user.repository.BadgeRepository;
+import server.user.repository.RefreshTokenRepository;
+import server.user.repository.UserRepository;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.anyLong;
+import static org.mockito.Mockito.times;
+import static server.user.entity.User.UserStatus.USER_QUIT;
+
+@ExtendWith(MockitoExtension.class)
+@SuppressWarnings("NonAsciiCharacters") // 한글 함수명을 쓰기 위함
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private RefreshTokenRepository refreshTokenRepository;
+
+    @Mock
+    private BadgeRepository badgeRepository;
+
+    @Mock
+    private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    @InjectMocks
+    private UserService userService;
+
+    private final String NON_EXISTED_EMAIL = "nonexisted@gmail.com";
+
+    @Test
+    void 유저_정보를_입력받으면_유저를_등록한다() {
+        // Given
+        UserPostRequest userPostRequest = UserPostRequest.builder()
+            .email("saview@gmail.com")
+            .password("saview!@#")
+            .nickname("saview")
+            .build();
+        given(bCryptPasswordEncoder.encode(anyString())).willReturn("encryptedPassword");
+        given(userRepository.save(any(User.class))).willAnswer(i -> i.getArguments()[0]);
+        given(badgeRepository.save(any(Badge.class))).willReturn(new Badge(any(User.class)));
+
+        // When
+        User createdUser = userService.createUser(userPostRequest);
+
+        // Then
+        assertThat(createdUser.getEmail()).isEqualTo("saview@gmail.com");
+        assertThat(createdUser.getPassword()).isEqualTo("encryptedPassword");
+        then(badgeRepository).should(times(1)).save(any(Badge.class));
+    }
+
+    @Test
+    void 입력한_비밀번호가_현재_비밀번호와_일치하면_새로운_비밀번호로_변경한다() {
+        // Given
+        String email = "saview@gmail.com";
+        User user = UserFixture.createUser(email);
+
+        given(userRepository.findByEmail(email)).willReturn(user);
+        given(bCryptPasswordEncoder.matches(anyString(), user.getPassword())).willReturn(true);
+
+        // When
+        userService.updatePassword(email, "saview!@#", "newSaview$%^");
+
+        // Then
+        then(bCryptPasswordEncoder).should(times(1)).encode("newSaview$%^");
+        then(userRepository).should(times(1)).findByEmail(email);
+    }
+
+    @Test
+    void 존재하는_이메일이면_해당_유저를_반환한다() {
```

Comment:
여기도 행위를 명시해주면 더 좋을 것 같아요.

```suggestion
    void 존재하는_이메일로_사용자_조회_시_해당_유저를_반환한다() {
```

Reply:
뭔가 어색했는데 행위가 빠졌었네요 ㅎㅎ..